tracer/test/Datadog.Trace.TestHelpers/SpanMetadataRules.cs

```diff
@@ -356,6 +356,7 @@ public static Result IsServiceStackRedis(this MockSpan span) => Result.FromSpan(
                 .Matches(Type, "redis"))
             .Tags(s => s
                 .IsPresent("redis.raw_command")
+                .IsPresent("db.redis.database_index")
```

Comment:
Isn't it optional?
```suggestion
                .IsOptional("db.redis.database_index")
```

Reply:
No, it should always be there in the case for ServiceStackRedis. They _always_ have a valid database index available, so we always attach it.

Reply:
Ok, I guess I'm missing something. The tests are failing saying `db.redis.database_index` is not present. But it's definitely there in the snapshots (which are passing). Does the spanmetadatarules not support metrics by any chance @zacharycmontoya? I assumed it did because the Kafka tests have:

```
                .IsOptional("message.queue_time_ms")
```

But I'm guessing that it's never actually finding that, and that we should create a separate `Metrics` asssertion instead?

Reply:
Oh I added the support for metrics
In one of the PRs I've closed
I'll try to find it for you

Reply:
See https://github.com/DataDog/dd-trace-dotnet/pull/3968